### PR TITLE
leaps 0.5.0 (new formula)

### PR DESCRIPTION
**Pull request by andyjeffries**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
### Description

Hopefully this all works - leap is a small Go utility that launches a webserver allowing you to collaborate on a codebase. https://github.com/Jeffail/leaps

### CODE REVIEW

File: Formula/leaps.rb [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        exec "#{bin}/leaps", "-address", port
+      end
+
+      # Give the server some time to start serving
+      sleep(1)
+
+      # Check that the server is responding correctly
+      output = `curl -o- http://localhost#{port}`
+      assert_match /Choose a document from the left to get started/, output
+      assert_equal 0, $?.exitstatus
```
Comment (ilovezfs) [inline Formula/leaps.rb]: You can replace the three lines above with
```
assert_match  /Choose a document from the left to get started/, shell_output("curl -o- http://localhost#{port}")
```

File: Formula/leaps.rb [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  def install
+    ENV["GOBIN"] = bin
+    ENV["GOPATH"] = buildpath
+    ENV["GOHOME"] = buildpath
+
+    mkdir_p buildpath/"src/github.com/jeffail/"
+    ln_sf buildpath, buildpath/"src/github.com/jeffail/leaps"
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    chdir buildpath
```
Comment (ilovezfs) [inline Formula/leaps.rb]: `cd buildpath` would be more idiomatic, but it turns out we're already in the `buildpath` directory here, so you can just remove the line.

**Comment by ilovezfs** [comment]
Is there any documentation you'd like to install, too? `doc.install ...`

**Comment by ilovezfs** [comment]
Otherwise, :+1:

**Comment by apjanke** [comment]
Minor audit failure.

```
Error: 2 problems in 1 formula
leaps:
  * C: 96: col 19: Unnecessary spacing detected.
  * C: 96: col 19: Put one space between the method name and the first argument.
```

Strangely, the audit passed on Mavericks, which may be why you didn't see this locally while testing.

**Comment by andyjeffries** [comment]
There's no documentation in the package @ilovezfs, if the author of the program releases any I'll add it.

@apjanke Thanks for spotting that. It was actually a line I copied and pasted from @ilovezfs and didn't re-run the audit check afterwards (rookie error!). Fixed it now, thank you.

**Comment by ilovezfs** [comment]
Looks good. Do you mind squashing this down to one commit?

**Comment by DomT4** [comment]
Are you sure we need the `go_resource`s here? It looks like upstream has them vendored so they shouldn't actually be necessary to declare explicitly in-formula.

**Comment by andyjeffries** [comment]
They're gitsubmodules, but the downloadable zip doesn't contain them.

**Comment by andyjeffries** [comment]
Squashed down to one commit now @ilovezfs. Thanks for all your help.

**Comment by DomT4** [comment]
> They're gitsubmodules, but the downloadable zip doesn't contain them, nor does it keep a .git folder so I can initialise them from within the extracted download.

We can do `git clones` in Homebrew as URLs. We have to do this kind of thing for `go`-using formulae fairly often so it wouldn't be an uncommon step:

```ruby
url "https://github.com/Jeffail/leaps.git",
    :tag => "v0.5.0",
    :revision => "5cf7328a8c498041d2a887e89f22f138498f4621"
```

**Comment by andyjeffries** [comment]
OK, thanks @DomT4, is there any easy step to then do a `git submodule update --init` - or do I just to it in the install step?  Or can I add a `--recursive` flag to the url command?

**Comment by DomT4** [comment]
It should happen automagically if `.gitmodules` exists after the initial `git clone` mechanism.

**Comment by andyjeffries** [comment]
I just tried it and get in to dependency hell again. For some reason it runs `git submodule update --init --recursive` but then fails when it can't find the dependencies, e.g.:

```
==> go build -o /usr/local/Cellar/leaps/0.5.0/bin/leaps ./cmd/leaps
cmd/leaps/bindata_assetfs.go:203:2: cannot find package "github.com/elazarl/go-bindata-assetfs" in any of:
	/usr/local/Cellar/go/1.6.2/libexec/src/github.com/elazarl/go-bindata-assetfs (from $GOROOT)
	/private/tmp/leaps-20160427-24905-xu55bj/src/github.com/elazarl/go-bindata-assetfs (from $GOPATH)
```

I would say for now, it works, can it just go in as is?  I assume the build system when it gets brought into master will make bottles for it, so rebuilding should be a rare case anyway, right? (and I know the build works with all the dependencies listed)

**Comment by DomT4** [comment]
Yeah, guess it's fine as-is in that case. Just would've been nice to avoid `resource`s if possible, people tend to forget to update 'em 😄. Thanks for checking it out!

**Comment by ilovezfs** [comment]
@andyjeffries What is the sequence of commands that works outside of brew? Maybe that will help clarify what's going wrong.

**Comment by andyjeffries** [comment]
@ilovezfs The set of steps is listed at https://github.com/Jeffail/leaps#vendoring but I wonder if Go is doing something funky when run interactively as opposed to when being run in a sandboxed environment like this (I'm an "OK" Go programmer, but not an expert yet so I'm not sure why it's breaking".

**Comment by andyjeffries** [comment]
I don't know whether I'll maintain the formula going forward or whether the application author will take over once I've got it in initially, but I'm happy enough updating all the dependencies when I update the formula to a new version of leaps.

**Comment by ilovezfs** [comment]
@andyjeffries I'm also fine with this coming in as is as I said above, but if you can figure out how to implement @DomT4's suggestion subsequently that would be awesome.

**Comment by DomT4** [comment]
I can play with it later if OP has had enough of Go for the day; I'd be sympathetic to that feeling 😉. Happy for this to ship as-is now though if you are.

**Comment by ilovezfs** [comment]
@DomT4 That would be lovely, since the author's vendoring doesn't look particularly sketchy or anything.

**Comment by ilovezfs** [comment]
@DomT4 I'll go ahead and pull as is then.

**Comment by andyjeffries** [comment]
I've raised it with the author of the app, but for now I'd love it to be merged in.  Thanks.

**Comment by ilovezfs** [comment]
@andyjeffries Thank you for your contribution to Homebrew core!

**Comment by andyjeffries** [comment]
No problem @ilovezfs - thanks for all your help and being so welcoming!

**Comment by andyjeffries** [comment]
(and thanks to @DomT4 too)

**Comment by ilovezfs** [comment]
@andyjeffries you're welcome :)

**Pull request closed without merging**